AWS/aws_helpers/sqs_wrapper.py

```diff
@@ -0,0 +1,131 @@
+
+        Raises:
+            ValueError: If queue url is not defined
+
+        Returns:
+            str:
+        """
+        result: str | None = None
+        if self._configuration and self._configuration.queue_url:
+            result = self._configuration.queue_url
+        else:
+            async with self.get_client("sqs") as sqs:
+                data = await sqs.get_queue_url(QueueName=self._configuration.queue_name)
+                result = data.get("QueueUrl")
+
+        if result:
+            return result
+
+        raise ValueError("Queue url is not defined")
+
+    @asynccontextmanager
+    async def receive_messages(
+        self, frequency: int | None = None, max_messages: int = 10, delete_consumed_messages: bool | None = None
+    ) -> AsyncGenerator[list[str], None]:
+        """
+        Receive SQS messages.
+
+        After processing messages they will be deleted from queue if delete_consumed_messages is True.
+
+        Example of usage:
+        with sqs.receive_messages() as messages:
+            # do something with messages
+
+        Args:
+            frequency: int
+            max_messages: int
+            delete_consumed_messages: int
+
+        Yields:
+            list[str]:
+        """
+        if max_messages < 1 or max_messages > 10:
+            raise ValueError("max_messages should be between 1 and 10")
+
+        frequency = frequency or self._configuration.frequency
+        delete_consumed_messages = delete_consumed_messages or self._configuration.delete_consumed_messages
+        queue_url = await self.queue_url()
+
+        async with self.get_client("sqs") as sqs:
+            try:
+                response = await sqs.receive_message(
+                    QueueUrl=queue_url,
+                    MaxNumberOfMessages=max_messages,
+                    WaitTimeSeconds=frequency,
+                    MessageAttributeNames=["All"],
+                    AttributeNames=["All"],
+                    VisibilityTimeout=0,
+                )
+
+            except Exception as e:  # pragma: no cover
+                logger.error(f"Failed to receive messages from sqs: {e}")
+
+                raise e
+
+            result = []
+
+            try:
+                for message in response.get("Messages", []):
+                    result.append(message["Body"])
+
+                logger.info(f"Received {len(result)} messages from sqs queue {self._configuration.queue_name}")
+
+                yield result
+            finally:
+                # We should delete messages from queue after releasing context manager if it is configured
+                if delete_consumed_messages and response.get("Messages", []):
+                    logger.info("Deleting consumed messages from sqs")
+                    for message in response.get("Messages", []):
+                        await sqs.delete_message(QueueUrl=queue_url, ReceiptHandle=message["ReceiptHandle"])
```

Comment:
Does the async sqs client not allow messages to be deleted in batch? Like [this code](https://github.com/SEKOIA-IO/automation-library/blob/main/AWS/aws/sqs/trigger_sqs_messages.py#L64-L67)

Reply:
As per docs from here https://github.com/aio-libs/aiobotocore#supported-aws-services `sqs` wrapper is only partially supported.
I was not able to find any tests that  proves that delete batch operation works as expected so I tried myself.
I was not able to use it, as I faced generic `Exception` with strange message.
I can investigate it more, or we can try to use custom implementation